mdata/chunk/sizes.go

```diff
@@ -0,0 +1,60 @@
+package chunk
+
+import (
+	"sync"
+)
+
+type Size uint8
+
+var ChunkSizes = map[Size]uint32{
+	1:  1,
+	2:  5,
+	3:  10,
+	4:  15,
+	5:  20,
+	6:  30,
+	7:  60,    // 1m
+	8:  90,    // 1.5m
+	9:  120,   // 2m
+	10: 180,   // 3m
+	11: 300,   // 5m
+	12: 600,   // 10m
+	13: 900,   // 15m
+	14: 1200,  // 20m
+	15: 1800,  // 30m
+	16: 2700,  // 45m
+	17: 3600,  // 1h
+	18: 5400,  // 1.5h
+	19: 7200,  // 2h
+	20: 9000,  // 2.5h
+	21: 10800, // 3h
+	22: 14400, // 4h
+	23: 18000, // 5h
+	24: 21600, // 6h
+	25: 25200, // 7h
+	26: 28800, // 8h
+	27: 32400, // 9h
+	28: 36000, // 10h
+	29: 43200, // 12h
+	30: 54000, // 15h
+	31: 64800, // 18h
+	32: 86400, // 24h
+}
+
+var RevChunkSizes = make(map[uint32]Size, len(ChunkSizes))
+
+var initMutex sync.Mutex
+var initialized = false
```

Comment:
It's not easy to find, but on https://golang.org/ref/spec#Package_initialization it says

> Package initialization—variable initialization and the invocation of init functions—happens in a single goroutine, sequentially, one package at a time. An init function may launch other goroutines, which can run concurrently with the initialization code. However, initialization always sequences the init functions: it will not invoke the next one until the previous one has returned.


Also it doesn't really matter much how many times a package is imported, each init function will only be called once.

Funny anecdote though.. we used to have a raintank fork of grafana, and do some symlinking tricks in our go packages.  I once made the mistake of importing the same package code from other packages, but using different import paths, in which case it ran the init twice since each was considered a different package, and this led to some really confusing results where both packages had different state.